src/app/participant-list/participant-list.component.ts

```diff
@@ -302,9 +307,13 @@ export class ParticipantListComponent implements OnInit {
                   } else if (question.questionType === "NUMERIC") {
                     type = Filter.NUMBER_TYPE;
                   }
-                  let displayName = this.getQuestionOrStableId( question );
-                  let filter = new Filter( new ParticipantColumn( displayName, question.stableId, activityDefinition.activityCode, null, true ), type, options );
-                  possibleColumns.push( filter );
+                  if (possibleColumns.find(filter => {
```

Comment:
extracting `if` expression as a variable with a meaningful name would be helpful for quick understanding in future.